app/src/main/java/org/cru/godtools/adapter/ToolsHeaderAdapter.kt

```diff
@@ -0,0 +1,61 @@
+package org.cru.godtools.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import org.ccci.gto.android.common.recyclerview.adapter.DataBindingViewHolder
+import org.cru.godtools.content.ToolHeader
+import org.cru.godtools.databinding.ToolsTutorialBannerBinding
+import org.cru.godtools.fragment.HeaderBannerCallbacks
+
+class ToolsHeaderAdapter internal constructor(builder: Builder) : BaseEmptyListHeaderFooterAdapter(builder) {
+    class Builder : BaseEmptyListHeaderFooterAdapter.Builder<Builder>() {
+        fun build(): ToolsHeaderAdapter {
+            return ToolsHeaderAdapter(this).apply {
+                this.adapterToolHeader = toolHeader
+            }
+        }
+
+        fun toolHeader(header: ToolHeader): Builder {
+            this.toolHeader = header
+            return this
+        }
+
+        lateinit var toolHeader: ToolHeader
+    }
+
+    // region Header
+
+    private lateinit var adapterToolHeader: ToolHeader
+
+    var isHeaderVisible: Boolean = true
+        set(value) {
+            field = value
+            headerAdapter?.notifyDataSetChanged()
+        }
+
+    lateinit var headerBannerCallbacks: HeaderBannerCallbacks
+
+    override fun getHeaderItemCount(): Int {
```

Comment:
you are missing implementations of `getHeaderItemViewType()` and `getHeaderItemId()`